Mathlib/Algebra/Order/Floor/Extended.lean

```diff
@@ -0,0 +1,252 @@
+/-
+Copyright (c) 2025 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Data.ENat.Lattice
+import Mathlib.Data.ENNReal.Order
+import Mathlib.Data.NNReal.Basic
+import Mathlib.Data.Real.ENatENNReal
+
+/-!
+# Extended floor and ceil
+
+This file defines the extended floor and ceil functions `ENat.floor, ENat.ceil : ℝ≥0∞ → ℕ∞`.
+
+## Main declarations
+
+* `ENat.floor r`: Greatest extended natural `n` such that `n ≤ r`.
+* `ENat.ceil r`: Least extended natural `n` such that `r ≤ n`.
+
+## Notation
+
+* `⌊r⌋ₑ` is `ENat.floor r`.
+* `⌈r⌉ₑ` is `ENat.ceil r`.
+
+The index `ₑ` is used in analogy to the notation for `enorm`.
+
+## TODO
+
+The day Mathlib acquires `ENNRat`, it would be good to generalise this file to an `EFloorSemiring`
+typeclass.
+
+## Tags
+
+efloor, eceil
+-/
+
+open Set
+open scoped ENNReal NNReal
+
+namespace ENNReal
+variable {r : ℝ≥0} {n : ℕ}
+
+@[simp, norm_cast] lemma natCast_le_coe : (n : ℝ≥0∞) ≤ r ↔ n ≤ r := by simp [← coe_le_coe]
```

Comment:
```suggestion
@[simp, norm_cast] lemma natCast_le_ofNNReal : (n : ℝ≥0∞) ≤ r ↔ n ≤ r := by simp [← coe_le_coe]
```
or is there precedent for not using this name for ENNReal?

Reply:
Yes, there is precedent: `ENNReal.coe_le_coe`